### PR TITLE
Version5 3

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/admin/GetAllProductsAdminController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
+        List<Product> allProducts = productService.getAll();
+        req.setAttribute("products", allProducts);
+        req.getRequestDispatcher("/WEB-INF/views/admin/allProductsAdmin.jsp")
+                .forward(req, resp);
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/admin/GetAllProductsAdminController.java]: I think this line is less than 100 characters long, so it should be in one line, as it's not a stream. I got a comment on a similar piece of code from mentors earlier) You have several places where req.getRequestDispatcher(...) looks like this, but I'll comment only here.

File: src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java]: I think we don't need a newline here.

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java [hunk 3]
```diff
@@ -30,7 +30,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         Long id = Long.valueOf(userId);
         Product product = productService.getById(id);
         shoppingCartService.addProduct(shoppingCart,product);
-        resp.sendRedirect(req.getContextPath() + "/all/items/bucket");
+        resp.sendRedirect(req.getContextPath() + "/shopping-cart/products/add");
     }
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java]: Remove the newline here

File: src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java]: Remove the newline here

File: src/main/java/com/internet/shop/controllers/order/GetOrderController.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+public class GetOrderController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/order/GetOrderController.java]: Remove the newline here

File: src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (AnaLanda) [inline src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java]: Remove the newline here

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        </tr>
+    </c:forEach>
+    <p>
+        <a href="${pageContext.request.contextPath}/products/all">back to all items page</a>
+    </p>
+</table>
+<p>
+    <a href="${pageContext.request.contextPath}/">back to main page</a>
+</p>
+
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp:46]: I think you can remove the newline here, too

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+<body>
+<h2>ALL USERS ORDERS LIST (ADMIN)</h2>
+<table border="1">
+    <tr>
+        <td>User ID</td>
+        <td>Order ID</td>
+        <td>Items</td>
+        <td>Details</td>
+        <td>Delete order</td>
+
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: Remove the extra newlines in jsp files

File: src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp [hunk 9]
```diff
@@ -27,22 +26,17 @@
...
-<%--<p>--%>
-<%--    <a href="${pageContext.request.contextPath}/checkout/order">checkout</a>--%>
-<%--</p>--%>
-    <button type="submit">checkout</button>
+<p>
+    <a href="${pageContext.request.contextPath}/products/all">back to all items page</a>
+</p>
 <p>
     <a href="${pageContext.request.contextPath}/">back to main page</a>
 </p>
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp:40]: Remove the newline

File: src/main/webapp/WEB-INF/views/cart/all.jsp [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+<body>
+<form method="get" action="${pageContext.request.contextPath}/checkout/order">
+    <h2>ITEMS IN YOUR BUCKET</h2>
+    <table border="1">
+        <tr>
+            <td>ID</td>
+            <td>Name</td>
+            <td>Price</td>
+            <td>Delete</td>
+
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/cart/all.jsp:16]: Remove the newline

File: src/main/webapp/WEB-INF/views/orders/detailOrder.jsp [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Title</title>
+</head>
+<body>
+<h2>DETAIL ORDER</h2>
```
Comment (AnaLanda) [inline src/main/webapp/WEB-INF/views/orders/detailOrder.jsp]:
```suggestion
<h2>ORDER DETAILS</h2>
```

**Review by AnaLanda — COMMENTED** [review]
Looks really good! You have a couple extra newlines you should remove)

File: src/main/java/com/internet/shop/controllers/admin/GetAllProductsAdminController.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/admin/GetAllProductsAdminController.java]: ok. thanks

File: src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java]: done

File: src/main/java/com/internet/shop/controllers/order/GetOrderController.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/GetOrderController.java]: done

File: src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java]: done

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp (hunk 8 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: done

File: src/main/webapp/WEB-INF/views/orders/detailOrder.jsp (hunk 11 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/orders/detailOrder.jsp]: ok

**Comment by Sofasmile** [comment]
давай для початку пофіксимо усі конфлікти що відображаються нижче

File: src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/admin/orders")
+public class GetOrderAdminController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java]:
```suggestion
public class GetAllOrdersAdminController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+import com.internet.shop.service.interfaces.ProductService;
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/shopping-cart/products/del")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java]:
```suggestion
@WebServlet("/shopping-cart/products/delete")
```

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/checkout/order")
+public class CheckoutController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/CheckoutController.java]: optional: you can rename to CompleteOrderController

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+import com.internet.shop.service.interfaces.OrderService;
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/checkout/order")
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/CheckoutController.java]:
```suggestion
@WebServlet("/order/checkout (or /order/complete)")
```

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/CheckoutController.java]: remove empty line

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+@WebServlet("/orders/delete")
+public class DeleteOrderController extends HttpServlet {
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java]:
```suggestion
        String orderId = req.getParameter("id");
```

File: src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java:26]: чим відрізняється цей метод від CheckoutController?

File: src/main/java/com/internet/shop/controllers/order/GetOrderController.java [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/orders")
+public class GetOrderController extends HttpServlet {
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/GetOrderController.java]:
```suggestion
public class GetUserOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java [hunk 20]
```diff
@@ -0,0 +1,35 @@
...
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        long orderId = Long.parseLong(id);
+        Order order = orderService.getById(orderId);
+        req.setAttribute("order", order);
+        List<Product> allProducts = productService.getAll();
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java]: а чому ми тут відображаємо список всіх продуктів?

File: src/main/java/com/internet/shop/controllers/product/DeleteProductController.java [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+@WebServlet("/products/delete")
+public class DeleteProductController extends HttpServlet {
+    private static Injector injector = Injector.getInstance("com.internet.shop");
+    private ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String userId = req.getParameter("id");
```
Comment (Sofasmile) [inline src/main/java/com/internet/shop/controllers/product/DeleteProductController.java]:
```suggestion
        String productId = req.getParameter("id");
```

File: src/main/webapp/web.xml [hunk 22]
```diff
@@ -3,4 +3,44 @@
          xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
          http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
          version="3.1">
+<!--    <servlet>-->
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: давай видалимо коментований код

File: src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java (hunk 12 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/admin/GetOrderAdminController.java]: done

File: src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java (hunk 13 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java]: done

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java (hunk 14 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/CheckoutController.java]: done

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java (hunk 15 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/CheckoutController.java]: done

File: src/main/java/com/internet/shop/controllers/order/CheckoutController.java (hunk 16 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/CheckoutController.java]: done

File: src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java (hunk 17 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/DeleteOrderController.java]: done

File: src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java (hunk 18 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/GetCheckoutController.java:26]: Ничем. Дублирование.

File: src/main/java/com/internet/shop/controllers/order/GetOrderController.java (hunk 19 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/GetOrderController.java]: done

File: src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java (hunk 20 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/order/ShowDetailsOrderController.java]: сорри. это были эксперименты. забыл убрать.

File: src/main/java/com/internet/shop/controllers/product/DeleteProductController.java (hunk 21 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/product/DeleteProductController.java]: done

File: src/main/webapp/web.xml (hunk 22 above)
Comment (vremere-a) [reply src/main/webapp/web.xml]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**